### PR TITLE
feat(frontend): about me

### LINKED ISSUE #1018: 🚀 [Feature] Frontend: About me section

### PR DESCRIPTION
**Pull request by Bettelstab**
## 🍰 Pullrequest
Adds the About Me section to the cockpit

### Issues
- fixes #1018

### Todo
- [x] Style availability select
- [x] Real data
- [x] Can edit availability
- [x] Can edit details
- [x] Can edit social media
- [x] Can edit name
- [x] Can edit introduction
- [x] Focus name after click on name or edit icon
- [x] Focus introduction after click on introduction
- [x] Toggle details edit clicking on box
- [x] Placeholders for details
- [x] Style plus button (round, green for active)
- [x] Social media links validation / generation => probably good for now
- [ ] Name as textarea, watch for height changes => separate issue
- [x] Correct scrollbars
- [x] mdi Icons when possible
- [x] Icon colors
- [x] Header grid positions
- [x] Text color dark theme
- [x] Dropdown background color
- [x] Mobile width
- [x] Remove unused items
- [x] Disable add button while waiting for server response
- [x] Tests

Problems:
- The snapshot for `AboutMe` is without content, I don't get how to fix it.
- Had to outcomment `AboutMe.test.ts` => "updates User with new availability" because it wouldn't work
- Same for `EditSocialMedia.test.ts` => "emits add-social event"

### CODE REVIEW

**Comment by ogerly** [comment]
are these special icons? 
otherwise we should use the one from vuetify.
https://vuetifyjs.com/en/features/icon-fonts/#usage

**Comment by Bettelstab** [comment]
I'm not sure about Hardy's icon source. We are using the Vuetify icons component though, just with our own SVGs.

**Comment by Bettelstab** [comment]
> are these special icons? otherwise we should use the one from vuetify. https://vuetifyjs.com/en/features/icon-fonts/#usage

I replaced all icons by the mdi ones where possible.

**Review by ogerly — APPROVED** [review]

File: frontend/src/components/cockpit/about-me/AboutMeView.vue [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        </v-select>
+
+        <template v-if="mode === 'view'">
+          <button class="name" @click="editName">
+            {{ props.name }}
+            <v-icon icon="mdi mdi-pencil-outline"></v-icon>
+          </button>
+          <button class="introduction" @click="editIntroduction">
+            {{ introduction }}
+          </button>
```
[the remark this answers is not in the document]
Comment (Bettelstab) [reply frontend/src/components/cockpit/about-me/AboutMeView.vue:32]: Functionality-wise it's a button, but I don't want the styles coming with `v-button`.

File: frontend/src/components/cockpit/about-me/AboutMeView.vue [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+          />
+          <input
+            ref="introductionInput"
+            :value="props.introduction"
+            :placeholder="introduction"
+            class="introduction"
+            name="introduction"
+            maxlength="25"
+            @change="updateIntroduction"
+          />
```
[the remark this answers is not in the document]
Comment (Bettelstab) [reply frontend/src/components/cockpit/about-me/AboutMeView.vue:50]: Same here, I don't want the styling, to get the feeling of in-place editing.

File: frontend/src/components/cockpit/about-me/AboutMeView.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply frontend/src/components/cockpit/about-me/AboutMeView.vue:32]: @ogerly alles, was konventionell mit HTML gelöst werden kann, unbedingt mit HTML umsetzen. Wenn ein `<button>` reicht - nimm einen `<button>` anstelle eines `<v-button>`. Gleiches gilt für `<input>` und `<v-input>`. Also erstmal HTML versuchen, dann HTML+CSS und als letzte Möglichkeit HTML, CSS + JavaScript.

File: frontend/src/components/cockpit/about-me/AboutMeView.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply frontend/src/components/cockpit/about-me/AboutMeView.vue:50]: @ogerly Vuetify ist sehr invasiv (CSS via `style=` Attribute) und es ist nicht framework-agnostisch und schlechter gewartet als alternative Lösungen, die von vielen Frameworks gleichzeitig genutzt werden. Also, wenn etwas ohne Vuetify geht, lieber kein Vuetify verwenden.

**Pull request merged**